src/test/java/com/google/step/youtube/DescriptionParserTest.java

```diff
@@ -1,12 +1,219 @@
 package com.google.step.youtube;
 
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
 import org.junit.Assert;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
-// @RunWith(JUnit4.class)
+@RunWith(JUnit4.class)
 public final class DescriptionParserTest {
 
+    /*                                                                   *
+     * ==================== GENERAL TESTS FOR PARSE ==================== *
+     *                                                                   */
+
+    /** String will return no matches from all regexs in parse() */
+    @Test
+    public void parse_MatchNone() {
+        String desc = "Check out our UPDATED version which has all the NEW ELEMENTS here:  https://youtu.be/rz4Dd1I_fX0"
+                        + "\nThe TEETH Song (Memorize Every Tooth): https://youtu.be/PI3hne8C8rU"
+                        + "\nDownload on ITUNES: http://bit.ly/12AeW99 ";
+
+        List<String> actual = DescriptionParser.parse(desc);
+        List<String> expected = Collections.emptyList();
+        Assert.assertEquals(expected, actual);
+    }
+
+    /** String will return matches from all regexs in parse() */
+    @Test
+    public void parse_MatchAll() {
+        String desc = "Get 10% off (save up to $44!) your own authentic Japanese snack box from "
+            + "Bokksu using my link: https://bit.ly/3fYbkZ5 and code FUNGBROS10";
+
+        List<String> actual = DescriptionParser.parse(desc);
+        List<String> expected = Arrays.asList("FUNGBROS10");
+        Assert.assertEquals(expected, actual);
+    }
+
+    /** String will only have matches from CODE_NO_QUOTES regex */
+    @Test
+    public void parse_CODE_NO_QUOTES() {
+        String desc = "Get 10% off (save up to $44!) your own authentic Japanese snack box from "
+            + "Bokksu using my link: https://bit.ly/3fYbkZ5 and code FUNGBROS10";
+
+        List<String> actual = DescriptionParser.parse(desc);
+        List<String> expected = Arrays.asList("FUNGBROS10");
+        Assert.assertEquals(expected, actual);
+    }
+
+    // TODO(margaret): when 2nd regex is added, test for allowing duplicate promocodes
+
+
+    /*                                                                   *
+     * ===================== TEST 'CODE' NO QUOTES ===================== *
+     *                         ex. "code OFF20"                          */
+
+    private final Pattern CODE_NO_QUOTES_PATTERN = DescriptionParser.Patterns.CODE_NO_QUOTES.getPattern();
+
+    /** CODE_NO_QUOTES: standard case - 1 space, end of string, all caps, letters and numbers */
+    @Test
+    public void codeNoQuotes_Standard() {
```

Comment:
optional: Why codeNoQuotes_  vs.  findMatches_.  Usually you name it: <methodUnderTest>_<state>
https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names

Reply:
CODE_NO_QUOTES is the name of the regex pattern in the DescriptionParser class. When I add more regexs later on, I want a way to differentiate between the tests since they will all be calling findMatches().